searching/floyd_cycle_detection_algorithm.c

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file floyd_cycle_detection_algorithm.c
+ * @brief Implementation of [Floyd's Cycle
+ * Detection](https://en.wikipedia.org/wiki/Cycle_detection) algorithm
+ * @details
+ * Given an array of integers containing 'n + 1' integers, where each
+ * integer is in the range [1, n] inclusive. If there is only one duplicate
+ * number in the input array, this algorithm returns the duplicate number in
+ * O(1) space and the time complexity is less than O(n^2) without modifying the
+ * original array, otherwise, it returns -1.
+ * @author [Swastika Gupta](https://github.com/Swastyy)
+ */
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c
#include <stdio.h>      /// for IO operations
#include <assert.h>    /// for assert
```